Library/CorrelationIdManager.ts

```diff
@@ -0,0 +1,93 @@
+import https = require('https');
+import http = require('http');
+import url = require('url');
+
+class CorrelationIdManager {
+    public static correlationIdPrefix: "cid-v1:";
+
+    private static pendingLookups: {[key: string]: Function[]} = {};
+    private static completedLookups: {[key: string]: string} = {};
+
+    public static queryCorrelationId(endpointBase: string, instrumentationKey: string, correlationIdRetryInterval: number, callback: (correlationId: string) => void) {

```

Comment:
It is possible for someone to construct multiple AI clients in the one node process/application with different instrumentation keys (or the same instrumentation key)